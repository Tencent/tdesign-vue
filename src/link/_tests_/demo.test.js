/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/examples/link/demos/base.vue';
import disabledVue from '@/examples/link/demos/disabled.vue';
import hoverVue from '@/examples/link/demos/hover.vue';
import iconVue from '@/examples/link/demos/icon.vue';
import sizeVue from '@/examples/link/demos/size.vue';
import themeVue from '@/examples/link/demos/theme.vue';
import underlineVue from '@/examples/link/demos/underline.vue';

const mapper = {
  baseVue,
  disabledVue,
  hoverVue,
  iconVue,
  sizeVue,
  themeVue,
  underlineVue,
};

describe('Link', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Link ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
