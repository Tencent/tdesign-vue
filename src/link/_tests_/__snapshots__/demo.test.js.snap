// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Link baseVue demo works fine 1`] = `
<div
  class="t-space t-space-horizontal"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-primary t-link--hover-underline"
      target=""
    >
      跳转链接
    </a>
  </div>
</div>
`;

exports[`Link Link disabledVue demo works fine 1`] = `
<div
  class="t-space t-space-horizontal"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-default t-is-disabled"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-primary t-is-disabled t-is-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-danger t-is-disabled"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-warning t-is-disabled"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-success t-is-disabled"
      target=""
    >
      
    跳转链接
  
      <span
        class="t-link__suffix-icon"
      >
        <svg
          class="t-icon t-icon-jump"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M2.3 13.7a1 1 0 00.7.3h10a1 1 0 001-1V8.5h-1V13H3V3h4.5V2H3a1 1 0 00-1 1v10c0 .27.1.52.3.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
          <path
            d="M9 3V2h4.5c.28 0 .5.22.5.5V7h-1V3.7L8.7 8 8 7.3 12.3 3H9z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;

exports[`Link Link hoverVue demo works fine 1`] = `
<div
  class="t-space t-space-vertical"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <div
      class="t-space t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-default t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-primary t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-danger t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-warning t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-success t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="t-space t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-default t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-primary t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-danger t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-warning t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-success t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="t-space t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-default t-is-underline t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-primary t-is-underline t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-danger t-is-underline t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-warning t-is-underline t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-success t-is-underline t-link--hover-color"
          target=""
        >
           跳转链接 
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Link Link iconVue demo works fine 1`] = `
<div
  class="t-space t-space-horizontal"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-default t-link--hover-underline"
      target=""
    >
      <span
        class="t-link__prefix-icon"
      >
        <svg
          class="t-icon t-icon-link"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.23 11.89l2.12-2.12.71.7-2.12 2.13A2.5 2.5 0 013.4 9.06l2.13-2.12.7.7-2.12 2.13a1.5 1.5 0 002.12 2.12zM10.47 9.06l-.7-.7 2.12-2.13a1.5 1.5 0 10-2.12-2.12L7.65 6.23l-.71-.7L9.06 3.4a2.5 2.5 0 013.54 3.54l-2.13 2.12z"
            fill="currentColor"
            fill-opacity="0.9"
          />
          <path
            d="M9.06 6.23L6.23 9.06l.71.7 2.83-2.82-.7-.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-primary t-is-underline t-link--hover-underline"
      href="https://tdesign.tencent.com/"
      target="_self"
    >
      <span
        class="t-link__prefix-icon"
      >
        <svg
          class="t-icon t-icon-link"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.23 11.89l2.12-2.12.71.7-2.12 2.13A2.5 2.5 0 013.4 9.06l2.13-2.12.7.7-2.12 2.13a1.5 1.5 0 002.12 2.12zM10.47 9.06l-.7-.7 2.12-2.13a1.5 1.5 0 10-2.12-2.12L7.65 6.23l-.71-.7L9.06 3.4a2.5 2.5 0 013.54 3.54l-2.13 2.12z"
            fill="currentColor"
            fill-opacity="0.9"
          />
          <path
            d="M9.06 6.23L6.23 9.06l.71.7 2.83-2.82-.7-.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      
    跳转链接
  
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-danger t-link--hover-color"
      href="https://tdesign.tencent.com/"
      target="_blank"
    >
      
    跳转链接
  
      <span
        class="t-link__suffix-icon"
      >
        <svg
          class="t-icon t-icon-jump"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M2.3 13.7a1 1 0 00.7.3h10a1 1 0 001-1V8.5h-1V13H3V3h4.5V2H3a1 1 0 00-1 1v10c0 .27.1.52.3.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
          <path
            d="M9 3V2h4.5c.28 0 .5.22.5.5V7h-1V3.7L8.7 8 8 7.3 12.3 3H9z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-warning t-is-disabled"
      target=""
    >
       跳转链接 
      <span
        class="t-link__suffix-icon"
      >
        <svg
          class="t-icon t-icon-jump"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M2.3 13.7a1 1 0 00.7.3h10a1 1 0 001-1V8.5h-1V13H3V3h4.5V2H3a1 1 0 00-1 1v10c0 .27.1.52.3.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
          <path
            d="M9 3V2h4.5c.28 0 .5.22.5.5V7h-1V3.7L8.7 8 8 7.3 12.3 3H9z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;

exports[`Link Link sizeVue demo works fine 1`] = `
<div
  class="t-space t-space-vertical"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <div
      class="t-space t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-s t-link--theme-default t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-s t-link--theme-primary t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-s t-link--theme-danger t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-s t-link--theme-warning t-is-disabled"
          target=""
        >
          跳转链接
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-s t-link--theme-success t-link--hover-underline"
          target=""
        >
          
      跳转链接
    
          <span
            class="t-link__suffix-icon"
          >
            <svg
              class="t-icon t-icon-jump"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M2.3 13.7a1 1 0 00.7.3h10a1 1 0 001-1V8.5h-1V13H3V3h4.5V2H3a1 1 0 00-1 1v10c0 .27.1.52.3.7z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M9 3V2h4.5c.28 0 .5.22.5.5V7h-1V3.7L8.7 8 8 7.3 12.3 3H9z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="t-space t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-default t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-primary t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-danger t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-warning t-is-disabled"
          target=""
        >
          跳转链接
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-m t-link--theme-success t-link--hover-underline"
          target=""
        >
          
      跳转链接
    
          <span
            class="t-link__suffix-icon"
          >
            <svg
              class="t-icon t-icon-jump"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M2.3 13.7a1 1 0 00.7.3h10a1 1 0 001-1V8.5h-1V13H3V3h4.5V2H3a1 1 0 00-1 1v10c0 .27.1.52.3.7z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M9 3V2h4.5c.28 0 .5.22.5.5V7h-1V3.7L8.7 8 8 7.3 12.3 3H9z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="t-space t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-l t-link--theme-default t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-l t-link--theme-primary t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-l t-link--theme-danger t-link--hover-underline"
          target=""
        >
           跳转链接 
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-l t-link--theme-warning t-is-disabled"
          target=""
        >
          跳转链接
        </a>
      </div>
      <div
        class="t-space-item"
      >
        <a
          class="t-link t-size-l t-link--theme-success t-link--hover-underline"
          target=""
        >
          
      跳转链接
    
          <span
            class="t-link__suffix-icon"
          >
            <svg
              class="t-icon t-icon-jump"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M2.3 13.7a1 1 0 00.7.3h10a1 1 0 001-1V8.5h-1V13H3V3h4.5V2H3a1 1 0 00-1 1v10c0 .27.1.52.3.7z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M9 3V2h4.5c.28 0 .5.22.5.5V7h-1V3.7L8.7 8 8 7.3 12.3 3H9z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Link Link themeVue demo works fine 1`] = `
<div
  class="t-space t-space-horizontal"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-default t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-primary t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-danger t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-warning t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-success t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
</div>
`;

exports[`Link Link underlineVue demo works fine 1`] = `
<div
  class="t-space t-space-horizontal"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-default t-is-underline t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-primary t-is-underline t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-danger t-is-underline t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-warning t-is-underline t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
  <div
    class="t-space-item"
  >
    <a
      class="t-link t-size-m t-link--theme-success t-is-underline t-link--hover-underline"
      target=""
    >
       跳转链接 
    </a>
  </div>
</div>
`;
